### ISSUE TITLE
reconstruct access list correctly in bus-mapping module

### ISSUE THREAD

**Issue by lispc** [scroll-tech/zkevm-circuits#171]
# Problem

a simple bytecode like `extcodesize 0xXXXX` may break the proof system due to incorrect access list calculated locally in statedb.rs.

# Solution 1

reconstruct access list correctly in bus-mapping module locally  @lightsing 

ref:
https://eips.ethereum.org/EIPS/eip-2929
https://github.com/ethereum/go-ethereum/blob/b3fc9574ecba5143ee1b61f172100c9228a04e18/core/vm/eips.go#L115
https://github.com/ethereum/go-ethereum/blob/b3fc9574ecba5143ee1b61f172100c9228a04e18/core/state/statedb.go#L1000

in short (impl by this order):

* extcodesize
* extcodecopy
* balance
* precompiled contract 
* tx sender, tx receiver (maybe done?)
* sstore sload (done)
* \*call\*  (done?) 
* optional access list in eip2930  (done?)

based on this branch: https://github.com/scroll-tech/zkevm-circuits/pull/164

# Solution 2 
Add it into StructLog directly: https://github.com/ethereum/go-ethereum/issues/25278

**Comment by lightsing** [comment]
## Memo

### Read Address Operations

add address into `accessed_addresses` and if target address not in `accessed_addresses`.

- `EXTCODESIZE`
- `EXTCODECOPY`
- `BALANCE`
- `CALL`
- `CALLCODE`
- `DELEGATECALL`
- `STATICCALL`

### Contract Creation

immediately add address to `accessed_addresses`, regardless of the result of `CREATE` or `CREATE2`.

### SLOAD

add into `accessed_storage_keys` if pair `(address, storage_key)` not in `accessed_storage_keys `

### SSTORE

add into `accessed_storage_keys` if pair `(address, storage_key)` not in `accessed_storage_keys `
